### DISCUSSION TITLE
multiple `<rehearsal>` children in a single `<direction-type>` element

### DISCUSSION THREAD

**Discussion by lemzwerg** [w3c-cg/musicxml#518] [Q&A] [answered]
For what exactly do I need multiple `<rehearsal>` children in a single `<direction-type>` element?  Is there an example?

I can imagine that this functionality is intended to provide font changes in rehearsal marks:

![image](https://github.com/w3c/musicxml/assets/1630245/506a5b13-9eb8-4214-b878-a88bda1f60bc)

In case my assumption is correct, however, I don't see any remark in the specification that the `enclosure` attribute should unify its arguments, so to say, preventing the following rendering.

![image](https://github.com/w3c/musicxml/assets/1630245/25da1fc1-e042-45fb-b319-6bc68a3eb5df)

Please advise.

**Comment by lemzwerg** [comment]
Exactly the same problem exists with `<words>` and `<symbol>`: it is not clear whether the `enclosure` attribute is intended to create a single enclosure around everything or multiple enclosures around each element.  Actually, I think that `enclosure` should have been added one level higher, i.e., as an attribute to `<direction-type>`.  Since it is too late to change this I suggest to add some explanatory notes.

**Reply by mscuthbert** [reply]
I definitely agree that enclosure should have either been a containing element or something with a list of ids or something it pertains to; maybe MNX can avoid repeating the problem. But as you say we'll need to look at where we are (what currently implementing systems expect and produce) and create guidelines and workarounds.

**Reply by lemzwerg** [reply]
I'm looking forward to these guidelines :slightly_smiling_face: 

Regarding MNX I think that a solution is quite simple: Either the enclosure attribute must be added to the parent element if there is more than a single child to enclose, or an enclosure container gets added, as you suggest.

**Comment by lemzwerg** [comment]
I've tested the snippet below with both MuseScore pre-4.4.0 and Finale 27.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE score-partwise PUBLIC "-//Recordare//DTD MusicXML 2.0 Partwise//EN"
                                "http://www.musicxml.org/dtds/partwise.dtd">
<score-partwise version="2.0">
  <part-list>
    <score-part id="P1">
      <part-name>MusicXML Part</part-name>
    </score-part>
  </part-list>
  <part id="P1">
    <measure number="1">
      <attributes>
        <divisions>1</divisions>
      </attributes>
      <direction>
        <direction-type>
          <rehearsal enclosure="rectangle">12</rehearsal>
          <rehearsal enclosure="rectangle" font-style="italic"> bis</rehearsal>
        </direction-type>
      </direction>
      <note>
        <pitch><step>C</step><octave>4</octave></pitch>
        <duration>4</duration>
        <type>whole</type>
      </note>
    </measure>
  </part>
</score-partwise>
```

Musescore: 
![image](https://github.com/w3c/musicxml/assets/1630245/a86b0158-24fc-4e68-a1d1-a91f5b4b2028)

Finale: 
![image](https://github.com/w3c/musicxml/assets/1630245/ab493de3-e9d9-4341-bca5-58646bb3b98d)

**Reply by lemzwerg** [reply]
It shouldn't make a difference if I remove the second `enclosure="rectangle"` statement – according to the MusicXML standard, this attribute gets inherited from the first `<rehearsal>` element.  MuseScore does it right; however, Finale produces a worse rendering result.  Looks like a bug...

![image](https://github.com/w3c/musicxml/assets/1630245/fe10b113-0b60-483c-90d3-a7818e53dea5)

**Reply by mscuthbert** [reply]
My thinking is that we will need an attribute "enclosure-break". Will give a full proposal when I'm back from a family trip next week.

**Reply by lemzwerg** [reply, marked as answer]
Great!  Meanwhile, I've filed #519.
